src/plugins/intel_gpu/src/graph/loop.cpp

```diff
@@ -375,16 +375,18 @@ loop_inst::concatenated_memory_mapping::ptr loop_inst::create_concat_memory_map(
     if (extern_mem_ptr != nullptr) {
         layout sliced_layout = intern_prim->get_output_layout(internal_id.idx);
         auto inter_mem_ptr = intern_prim->output_memory_ptr(internal_id.idx);
-        if (inter_mem_ptr == nullptr) {
+        if (inter_mem_ptr == nullptr || shape_changed()) {
             // if inner body intern_prim has no output memory because it has dynamic shape,
             // calculate inner body intern_prim layout using concat_mem's layout.
             auto updated_sliced_layout = sliced_layout.get_partial_shape();
             OPENVINO_ASSERT(updated_sliced_layout[io_prim_map.axis].is_static() || num_iterations > 0,
                                     "Not allowed dynamic dimension for axis when num_iteraiont is negative");
+
+            auto origin_input_pshape = body_network->get_input_node_output_layout(internal_id.pid).get_partial_shape();
```

Comment:
Why don't you use the following sample code to get the original input partial shape without creating the additional get_input_node_output_layout?

```suggestion
            auto origin_input_pshape = body_network->get_primitive(internal_id.pid)->get_node_output_layout();
```

Reply:
updated